src/megaapi_impl.cpp

```diff
@@ -1493,6 +1502,117 @@ char *MegaApiImpl::getBlockedPath()
+    {
+        LocalPath p = LocalPath::fromPath(localPath, *client->fsaccess);
+        LocalPath l = p.leafName();
+        backupName = l.toPath(*client->fsaccess);
+        request->setName(backupName.c_str()); // use this in putnodes_result()
+    }
+
+#if defined(_WIN32) && !defined(WINDOWS_PHONE)
+    if (!PathIsRelativeA(localPath.c_str()) && ((localPath.size() < 2) || localPath.compare(0, 2, "\\\\")))
+    {
+        localPath.insert(0, "\\\\?\\");
+        request->setFile(localPath.c_str());
+    }
+#endif
+
+    // get current user
+    User* u = client->ownuser();
+
+    // get handle of remote "My Backups" folder, from user attributes
+    if (!u || !u->isattrvalid(ATTR_MY_BACKUPS_FOLDER)) { return API_EACCESS; }
+    const string* handleContainerStr = u->getattr(ATTR_MY_BACKUPS_FOLDER);
+    if (!handleContainerStr) { return API_EACCESS; }
+
+    std::unique_ptr<TLVstore> tlvRecords(TLVstore::containerToTLVrecords(handleContainerStr, &client->key));
+    if (!tlvRecords || !tlvRecords->find("h")) { return API_EINTERNAL; }
+
+    handle h = 0; // make sure top two bytes are 0
+    memcpy(&h, tlvRecords->get("h").c_str(), MegaClient::NODEHANDLE);
+
+    if (!h || h == UNDEF) { return API_ENOENT; }
+
+    // get Node of remote "My Backups" folder
+    Node* myBackupsNode = client->nodebyhandle(h);
+    if (!myBackupsNode) { return API_ENOENT; }
+
+    // get 'device-id'
+    const string& deviceId = client->getDeviceid();
+    if (deviceId.empty()) { return API_EINCOMPLETE; }
+
+    // get `DEVICE_NAME`, from user attributes
+    if (!u->isattrvalid(ATTR_DEVICE_NAMES)) { return API_EINCOMPLETE; }
+    const string* deviceNameContainerStr = u->getattr(ATTR_DEVICE_NAMES);
+    if (!deviceNameContainerStr) { return API_EINCOMPLETE; }
+
+    tlvRecords.reset(TLVstore::containerToTLVrecords(deviceNameContainerStr, &client->key));
+    if (!tlvRecords || !tlvRecords->find(deviceId)) { return API_EINCOMPLETE; }
+
+    const string& deviceName = tlvRecords->get(deviceId);
+    if (deviceName.empty()) { return API_EINCOMPLETE; }
+
+    vector<NewNode> newnodes;
+    nameid attrId = AttrMap::string2nameid("dev-id"); // "device-id" would be too long
+    std::function<void(AttrMap& attrs)> addAttrsFunc = [=](AttrMap& attrs)
+    {
+        attrs.map[attrId] = deviceId;
+    };
+
+    // serch for remote folder "My Backups"/`DEVICE_NAME`/
+    Node* deviceNameNode = client->childnodebyname(myBackupsNode, deviceName.c_str(), false);
+    if (deviceNameNode) // validate this node
+    {
+        if (deviceNameNode->type != FOLDERNODE) { return API_EACCESS; }
+
+        // get 'dev-id' tag
+        const auto& childAttrs = deviceNameNode->attrs.map;
+        auto found = childAttrs.find(attrId);
+
+        // a folder without the attribute, or with it but with different value, will not be accessed
+        if (found == childAttrs.end() || found->second != deviceId) { return API_EACCESS; }
+    }
+    else // create `DEVICE_NAME` remote dir
```

Comment:
Since the device nodes can be renamed, shouldn't we be iterating the subfolders of "My Backups" and looking for one with the matching dev-id, since that seems to be our definition of matching the device with the folder?  thanks

Reply:
Good point. This should probably be changed to use your suggestion. @sergiohs84 what do you think?

Reply:
Matt is right. The `<device_name>` folder should be found among children of `My Backups`, but by checking the device id in the node's attribute, not by checking the name.
Maybe a getter `MegaHandle getDeviceNameFolder()` might be useful.

Reply:
Ok, I'll make that change

Reply:
Just for your information: 
- if the user changes the device name for a device id, the folders won't be renamed.
- if the user renames the `<device_name>` folder, the device name in the user attribute won't be updated.

Reply:
> Maybe a getter MegaHandle / MegaNode getDeviceNameFolder() might be useful.

I made the change in a slightly different way, because of the different error codes that could be encountered on the way.